### DISCUSSION TITLE
Usage of Yield Per with hybrid_property

### DISCUSSION THREAD

**Discussion by iKlotho** [sqlalchemy/sqlalchemy#12105] [Usage Questions] [answered]
Hello,

I have a question about using `yield_per` with `hybrid_property`. When I use `yield_per` alongside a `hybrid_property`, I encounter the error `Commands out of sync; you can't run this command now`. Specifically, I don't preload the required fields for the `hybrid_property`, so each yielded instance has to execute a separate query to fetch the necessary data.

I can resolve this issue by loading the required fields beforehand, but I'm curious about the root cause of the problem. Interestingly, the error occurs only when the `yield_per` size is smaller than the total row count of the query results. When I increase the `yield_per` size to be more than the number of rows returned by the query, the issue disappears, and I can query additional data from the ORM object without errors.

I’ve created a simple script to reproduce this issue. Running it against MySQL should trigger the error. However, the problem does not occur when I run the script using SQLite/PostgreSQL as the database.

I am using sqlalchemy version 2.0.36.

```python
from sqlalchemy import create_engine, Column, Integer, String
from sqlalchemy.ext.hybrid import hybrid_property
from sqlalchemy.orm import declarative_base, sessionmaker, load_only


engine = create_engine('mysql://user:pass@127.0.0.1:4308/test', echo=False)
# no problem with sqlite
# engine = create_engine("sqlite:///test.db", echo=False)
Base = declarative_base()

class User(Base):
    __tablename__ = 'users'

    id = Column(Integer, primary_key=True)
    name = Column(String(50), nullable=False)
    email = Column(String(120), unique=True, nullable=False)
    age = Column(Integer)

    def __repr__(self):
        return f"<User(name='{self.name}', email='{self.email}', age={self.age})>"

    @hybrid_property
    def random_property(self):
        return f"{self.name}+{self.age}"

Base.metadata.create_all(engine)
Session = sessionmaker(bind=engine)
session = Session()

def load_data(row_count):
    for i in range(row_count):
        user = User(name='John Doe', email=f'john_{i}@example.com', age=30)
        session.add(user)
    try:
        session.commit()
    except:
        session.rollback()

row_count = 100
# problem disappears when yield_size > row_count
yield_size = 10
load_data(row_count )
all_users = session.query(User).options(load_only(User.email)) # loading only the mail
for user in all_users.yield_per(yield_size):
    # This sends an additional query to obtain `name` and `age`
    print("user hybrid_property access", user.random_property)
```

**Comment by CaselIT** [comment, marked as answer]
Hi,

Some drivers/database support multiple results at the same time. Mysql usually can not.
What is happening here is that with yield per not all results are loaded at once, and if you then run another query you end up with multiple open results, that may cause issues depending on the driver/database

**Reply by iKlotho** [reply]
Thanks. I think `hybrid_property` is not a good choice if you plan to use `yield_per` with MySQL. The combination works only if you always include the required fields in the query, but it’s easy to forget.

Why can I send additional SQL queries without an error if the yield size exceeds the row count? Is it because there is an extra buffer to handle these additional queries?

**Reply by CaselIT** [reply]
> Why can I send additional SQL queries without an error if the yield size exceeds the row count?

In that case the results is closed in the first iteration, like if you did not use yield per
